packages/web3-eth-ens/src/ens.ts

```diff
@@ -0,0 +1,160 @@
+import { Address } from 'web3-utils';
+import { RevertInstructionError } from 'web3-common';
+import { NonPayableCallOptions, TransactionReceipt } from 'web3-eth-contract';
+import { Registry } from './registry';
+import { registryAddress } from './config';
+
+export class ENS {
+	public _registryAddress: string;
+	private readonly registry: Registry;
+
+	public constructor(registryAddr?: string) {
+		this.registry = new Registry(registryAddr);
+		this._registryAddress = registryAddr ?? registryAddress; // TODO change this when eth.net is finished
+	}
+
+	/**
+	 * Returns the Resolver by the given address
+	 */
+	public async getResolver(name: string): Promise<unknown> {
+		return this.registry.getResolver(name);
+	}
+
+	/**
+	 * set the resolver of the given name
+	 */
+	public async setResolver(
+		name: string,
+		address: Address,
+		txConfig: NonPayableCallOptions,
+	): Promise<TransactionReceipt | RevertInstructionError> {
+		return this.registry.setResolver(name, address, txConfig);
+	}
+
+	/**
+	 * Sets the owner, resolver and TTL for a subdomain, creating it if necessary.
+	 */
+	public async setSubnodeRecord(
+		name: string,
+		label: string,
+		owner: Address,
+		resolver: Address,
+		ttl: number,
+		txConfig: NonPayableCallOptions,
+	): Promise<TransactionReceipt | RevertInstructionError> {
+		return this.registry.setSubnodeRecord(name, label, owner, resolver, ttl, txConfig);
+	}
```

Comment:
Anything that we are setting, will not return `PromiEvent` instead of `Promise`?